src/bepasty/utils/date_funcs.py

```diff
@@ -31,6 +35,8 @@ def time_unit_to_sec(value, unit):
         'YEARS': 60 * 60 * 24 * 365,
         'FOREVER': FOREVER,
     }
+    if units.get(unit) is None:
+        raise BadRequest(description='Maxlife-Unit header is incorrect')
```

Comment:
this function does not really know what it is used for, so it should rather raise a KeyError and let the caller generate the BadRequest.

Reply:
hm, time_unit_to_sec() is only used from get_maxlife() if exclude tests/. it becomes like following. this was your intent?

```python
try:
    get_maxlife(...)
except KeyError:
     raise BadRequest(description='Maxlife-Unit header is incorrect')
```

Reply:
yes, or use some variable, if that can have underscore alternatively.

Reply:
ah, ok. changed the get_maxlife() to catch KeyError, and time_unit_to_sec() keep as is (KeyError is raised naturally if no key)

```python
    try:
        return time_unit_to_sec(value, unit)
    except KeyError:
        field = unit_key if underscore else 'Maxlife-Unit'
        raise BadRequest(description='{} header is incorrect'.format(field))
```

Reply:
ok